### ISSUE TITLE
Historical Variance Decomposition

### ISSUE THREAD

**Issue by htv399** [statsmodels/statsmodels#9441]
#### Historical Variance Decomposition VAR 
I was attempting to perform a Historical Variance Decomposition (HVD) using the VAR functions available in Statsmodels. However, to the best of my knowledge, I was unable to find a built-in function for this purpose. If there is currently a way to achieve this using Statsmodels, I would greatly appreciate your guidance.

To compute the HVD, I adapted the function from Ambrogio Cesa-Bianchi's VAR Toolbox for MATLAB and implemented it in Python. My solution is currently limited to handling endogenous variables, while exogenous variables are incorporated as part of the residuals.

If Statsmodels is still working on a native function for performing HVD, you can find my adapted code on my GitHub repository. It might provide a useful starting point or a shortcut for implementing this functionality: https://github.com/htv399/VARHD/tree/main

**Comment by josef-pkt** [comment]
What is the license of the matlab toolbox?

We can only use translated code if it is license compatible with our BSD-3 license

**Comment by htv399** [comment]
> What is the license of the matlab toolbox?
> 
> We can only use translated code if it is license compatible with our BSD-3 license

This is the link to the VAR matlab toolbox repository developed by Cesa-Bianchi https://github.com/ambropo/VAR-Toolbox. There you can find all the information related to it.

**Comment by josef-pkt** [comment]
It's GPL which is not license compatible with BSD/MIT.
So, we cannot use, translate that code.

If it's just some formulas, then it wouldn't be a problem. Coded formulas are not copyrightable.

**Comment by htv399** [comment]
Indeed, the Historical Variance Decomposition in Vector Autoregressive (VAR) models can be calculated using the various outputs generated by the statsmodel’s VAR function. I hope that you will be able to make these updates soon, as it is one of the most fascinating applications of this model. Please note that similar functionalities for VAR estimation, including the calculation and graphical representation of Historical Variance Decomposition, are already integrated into the MATLAB and R packages.
